### PR TITLE
Add hidden fields option to sanitizing

### LINKED ISSUE #4342: Hide / make private timestamps, _id, id and __v with MongoDB
**Describe the bug**
There is no way with the beta to make private some of the attributes through the model's JSON

**Steps to reproduce the behavior**
- Create a a new model with timestamps for example.
- Try to add it manually to the attributes in the model's JSON.
- Make it private.
- The API response will still contain the timestamps.
(Also valid with id, _id, and __v)

**Expected behavior**
We should be able to hide those attributes from the API response

**System**
- Node.js version: v11.3.0
- NPM version: 6.5.0
- Strapi version: v3.0.0-beta.17.2
- Database: Mongodb
- Operating system: MacOS/Debian

### PR DESCRIPTION
**Pull request by lambertkevin**
#### Description of what you did:
Added a `hiddenFields` attribute to the response config in order to sanitize entities globally. It will help preventing default values from appearing on every responses of every entities. (e.g. `__v` or timestamps when using mongodb).

Fix #4342 

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [X] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [X] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [X] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
As a side note @lambertkevin because I know the Strapi team will say it, can you please add documentation for this?

**Comment by alexandrebodin** [comment]
THanks for your contribution. I’ll take a look at i later.

File: packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json [hunk 1]
```diff
@@ -8,5 +8,11 @@
   "poweredBy": {
     "enabled": true,
     "value": "Strapi <strapi.io>"
-  }
+  },
+  "hiddenFields": [
```
Comment (lauriejim) [inline packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json]: I'm not sure we have to force `SQL` apps  (MySQL, SQLite, Postgres) to have these values by default.

Comment (derrickmehaffy) [reply packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json]: I do agree about removing the timestamp ones, those are very useful (That being said SQL use `created_at` and `updated_at`)

**Comment by alexandrebodin** [comment]
The `hiddenFields` param shouldn't be set on the configuration files by default. A good documentation with an example would  work just fine.

File: packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json (hunk 1 above)
Comment (lambertkevin) [reply packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json]: Ok I just removed the default values from the configs.

**Comment by lambertkevin** [comment]
@alexandrebodin Ok removed the default configs and added example to the documentation

**Comment by alexandrebodin** [comment]
Thanks for the update. I meant remove the `hiddenFields` key from the files completly.

File: packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json [hunk 2]
```diff
@@ -8,5 +8,6 @@
   "poweredBy": {
     "enabled": true,
     "value": "Strapi <strapi.io>"
-  }
+  },
+  "hiddenFields": []
```
Comment (derrickmehaffy) [inline packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json]: Remove this hiddenFields from the config

Comment (derrickmehaffy) [reply packages/strapi-generate-new/lib/resources/files/config/environments/production/response.json]: @victorlambert

**Comment by alexandrebodin** [comment]
Hi @lambertkevin do you think you will be able to make the last changes ? Thanks !

**Comment by alexandrebodin** [comment]
Hi @lambertkevin I'm going to close this PR for now until you can make the changes ;).

**Pull request closed without merging**